optimum/intel/openvino/quantization.py

```diff
@@ -344,73 +348,69 @@ def _quantize_torchmodel(
             )
-            quantization_config = OVConfig(compression=INT8_WEIGHT_COMPRESSION_CONFIG) if weights_only else OVConfig()
-
-        model_inputs = next(iter(calibration_dataloader))
-        quantization_config.add_input_info(model_inputs)
-        nncf_config = NNCFConfig.from_dict(quantization_config.__dict__)
-        nncf_config = register_default_init_args(nncf_config, calibration_dataloader)
-        controller, compressed_model = create_compressed_model(
-            self.model, nncf_config, wrap_inputs_fn=wrap_nncf_model_inputs_with_objwalk
+            quantization_config = OVConfig()
+        onnx_file_name = (
+            ONNX_WEIGHTS_NAME
+            if file_name is None and quantization_config.save_onnx_model
+            else Path(ov_file_name).with_suffix(".onnx")
         )
-        compressed_model = controller.strip(do_copy=False)
+        if weights_only:
+            compressed_model = compress_weights(self.model)
+            self.model = compressed_model
+        else:
+            calibration_dataloader = self._get_calibration_dataloader(
+                calibration_dataset=calibration_dataset,
+                batch_size=batch_size,
+                remove_unused_columns=remove_unused_columns,
+                data_collator=data_collator,
+            )
+
+            model_inputs = next(iter(calibration_dataloader))
+            quantization_config.add_input_info(model_inputs)
+            nncf_config = NNCFConfig.from_dict(quantization_config.__dict__)
+            nncf_config = register_default_init_args(nncf_config, calibration_dataloader)
+            controller, compressed_model = create_compressed_model(
+                self.model, nncf_config, wrap_inputs_fn=wrap_nncf_model_inputs_with_objwalk
+            )
+            compressed_model = controller.strip(do_copy=False)
 
         task = self.task
         model = self.model
         self.model.config.save_pretrained(save_directory)
-
+        model = patch_decoder_attention_mask(model)
         if task == "text-generation":
             onnx_config = onnx_config_class(model.config, use_past=model.config.use_cache)
         else:
             onnx_config = onnx_config_class(model.config)
 
-        onnx_path = save_directory / ONNX_WEIGHTS_NAME
-
-        # Export the model to the ONNX format
+        model_path = save_directory / (onnx_file_name if quantization_config.save_onnx_model else ov_file_name)
+        onnx_path = save_directory / onnx_file_name
+        export_fn = export if not quantization_config.save_onnx_model else export_pytorch_via_onnx
         opset = min(onnx_config.DEFAULT_ONNX_OPSET, MAX_ONNX_OPSET)
         opset = max(opset, MIN_ONNX_QDQ_OPSET)
-        export(
-            model=compressed_model,
-            config=onnx_config,
-            opset=opset,
-            output=onnx_path,
-        )
+        _, _, is_onnx = export_fn(model=model, config=onnx_config, output=model_path, opset=opset)
+        if is_onnx:
```

Comment:
Is it equivalent to :   
```suggestion
        if quantization_config.save_onnx_model:
```

Reply:
no, it is not equivalent, if something went wrong and conversion got fallback to onnx, this case will not triggered